docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set.md

```diff
@@ -0,0 +1,112 @@
+
+
+### Upgrading from Pet Sets to Stateful Sets
+
+#### Find all Pet Sets and their manifests 
+
+First, find all existing Pet Sets in your cluster:
+
+```shell
+kubectl get petsets --all-namespaces
+```
+
+If you don't find any existing Pet Sets, you can safely upgrade your cluster to >= 1.5.
+
+If you find existing Pet Sets and you have all their manifests at hand, you can continue to the next step to prepare Stateful Set manifests. 
+
+Otherwise, you need to save their manifests so that you can recreate them as Stateful Sets later. 
+Here's an example command for you to save all existing Pet Sets as one file. 
+
+```shell
+# Save all existing Pet Sets in all namespaces into a single file. Only needed when you don't have their manifests at hand. 
+kubectl get petsets --all-namespaces -o yaml > all-petsets.yaml
+```
+
+#### Prepare Stateful Set manifests 
+
+Now, for every Pet Set manifest you have, prepare a corresponding Stateful Set manifest with the following steps: 
+
+1. Change `apiVersion` from `apps/v1alpha1` to `apps/v1beta1`
+2. Change `kind` from `PetSet` to `StatefulSet`
+3. If you have the debug hook annotation `pod.alpha.kubernetes.io/initialized` set to `true`, you may remove it since it's redundant. If you don't have this annotation, you should add one, with the value set to `false`, to pause Stateful Sets operations.
+
+It's recommended that you keep both Pet Set manifests and Stateful Set manifests, so that you may safely rollback and recreate your Pet Sets, 
+if you decide not to upgrade your cluster. 
+
+#### Delete all Pet Sets without cascading
+
+If you find existing Pet Sets in your cluster in the previous step, you need to delete all Pet Sets *without cascading*. You can do this from `kubectl` with `--cascade=false`. 
+Note that if the flag wasn't set, **cascading deletion will be performed by default**, and all pods managed by your Pet Sets will be gone. 
+
+```shell
+# Delete all existing Pet Sets without cascading 
+kubectl delete petsets -f <pet-set-file> --cascade=false
+
+# Alternatively, delete them by name and namespace without cascading
+kubectl delete petsets <pet-set-name> -n=<pet-set-namespace> --cascade=false
+
+# Get all Pet Sets again to make sure you deleted them all 
+kubectl get petsets --all-namespaces
+```
+
+At this moment, you've deleted all Pet Sets in your cluster, but not their pods, persistent volumes, or persistent volume claims. 
+However, since the pods are not managed by Pet Sets anymore, they will be vulnerable to node failures until you finish the upgrade and recreate Stateful Sets.
+
+#### Upgrade your Kubernetes cluster to >= 1.5
+
+Now, you may [upgrade your Kubernetes cluster](/docs/admin/cluster-management/#upgrading-a-cluster), or just your master, to >= 1.5.0.
```

Comment:
You're right. They shouldn't upgrade nodes at this point, otherwise all pods will be gone.